pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -780,6 +783,46 @@ func lookupDomainConfig(ctx *domainContext, key string) *types.DomainConfig {
 	return &config
 }
 
+func execWithTimeout(command string, args ...string) ([]byte, bool, error) {
+
+	ctx, cancel := context.WithTimeout(context.Background(),
+		10*time.Second)
+	defer cancel()
+
+	cmd := exec.CommandContext(ctx, command, args...)
+	out, err := cmd.Output()
+	if ctx.Err() == context.DeadlineExceeded {
+		return nil, false, nil
+	}
+	return out, true, err
+}
+
+// Execute rkt commands
+func ExecuteRktCmd(URL string) []byte {
+
+	cmd := "rkt"
+	arg1 := "run"
+	//arg2 := "registry-1.docker.io/zededa/zcli:latest"
+	arg2 := URL
```

Comment:
Ah! I see what you're doing -- you're trying to combine fetch and run into a single invocation. While this is very tempting for the initial implementation as @nordmark mentioned -- we have to split the downloader and domainmanager parts from the get-go

Reply:
Did not realize that passing url tries to do fetch again... We are doing fetch before run - so run should find the image locally.. As discussed, we will move to using SHA

Reply:
Based on rkt documentation, if the image is found locally then it is not downloaded again.
Am I missing something?

Reply:
There are certain conditions under which it will happen, I believe.

But the point is moot -- I don't think I can see us merging code that has the registry URL